core/src/test/scala/unit/kafka/log/remote/RemoteIndexCacheTest.scala

```diff
@@ -117,51 +128,60 @@ class RemoteIndexCacheTest {
 
   @Test
   def testCacheEntryExpiry(): Unit = {
-    val cache = new RemoteIndexCache(maxSize = 2, rsm, logDir = logDir.toString)
+    // close existing cache created in test setup before creating a new one
+    Utils.closeQuietly(cache, "RemoteIndexCache created for unit test")
```

Comment:
If we don't do this at the beginning of the test, will it fail the tests? I thought we've already done it at the `AfterEach`

Reply:
No, it wouldn't fail the test but it will lead to memory & thread leaks. This is because in some tests, we are forcefully assigning a new value to member variable `cache` in the test. Hence, we end up with two instances of cache, one which was initialized in `@BeforeEach` and another which the test created by itself. If we don't explicitly close the cache  which is initialized at the `@BeforeEach`, it will keep on consuming heap and have stray threads which won't be released until process is closed.

Reply:
OK, got it. Make sense.